File: mobile/src/main/java/com/gitlab/PCU/PCU/helper/MaxLineFilter.java
package com.gitlab.PCU.PCU.helper;

import android.text.InputFilter;
import android.text.Spanned;

/**
 * Created by tim on 28.01.17.
 */

public class MaxLineFilter implements InputFilter {
    private int maxLines;

    public MaxLineFilter(int maxLines) {
        this.maxLines = maxLines;
    }

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        int lineCount = 0;
        for (char c : source.toString().toCharArray()) {
            if (c == '\n') {
                lineCount++;
            }
        }
        for (char c : dest.toString().toCharArray()) {
            if (c == '\n') {
                lineCount++;
            }
        }
        if ((lineCount + 1) <= maxLines) {
            return null;
        } else {
            return "";
        }
    }
}

File: mobile/src/main/java/com/github/PCUnlocker/app/helper/MaxLineFilter.kt
package com.github.PCUnlocker.app.helper

import android.text.InputFilter
import android.text.Spanned

/**
 * Created by tim on 28.01.17.
 */

class MaxLineFilter(private val maxLines: Int) : InputFilter {

    override fun filter(source: CharSequence, start: Int, end: Int, dest: Spanned, dstart: Int, dend: Int): CharSequence? {
        var lineCount = 0
        for (c in source.toString().toCharArray()) {
            if (c == '\n') {
                lineCount++
            }
        }
        for (c in dest.toString().toCharArray()) {
            if (c == '\n') {
                lineCount++
            }
        }
        if (lineCount + 1 <= maxLines) {
            return null
        } else {
            return ""
        }
    }
}
